src/ThundraLogger.ts

```diff
@@ -2,39 +2,35 @@ import ConfigProvider from './config/ConfigProvider';
 import ConfigNames from './config/ConfigNames';
 
 class ThundraLogger {
-    static instance: ThundraLogger;
-    debugEnabled: boolean;
 
-    constructor() {
-        this.debugEnabled = ConfigProvider.get<boolean>(ConfigNames.THUNDRA_DEBUG_ENABLE);
-        ThundraLogger.instance = this;
-    }
+    static debugEnabled: boolean;
 
-    static getInstance(): ThundraLogger {
-        return ThundraLogger.instance ? ThundraLogger.instance : new ThundraLogger();
+    private constructor() {
```

Comment:
I didn't see where this constructor is called, so if this is the case, I think `static debugEnabled: boolean` field is always equal to the default value `undefined`.

Reply:
Yes, it was not called. Also since static initializer of `ThundraLogger` is called before `ConfigProvider` is initialized, I updated debug enable check by looking at config on `isDebugEnabled` method call. Otherwise, `ThundraLogger` was not getting the right config value and I also didn't want to tweak with the module initialization order

Reply:
Also keep private constructor to prevent `ThundraLogger` from being instantiated

Reply:
👍